### PR TITLE
Subscriptions: fixing fallback after local platform failure

### PR DESCRIPTION
**Pull request by prateekbh**
- Shifts resolution of fallback entitlement right after local platform fails

### CODE REVIEW

**Comment by jpettitt** [comment]
Can you add a test for the case where non-local fails to make sure it doesn't use fallback.

File: extensions/amp-subscriptions/0.1/platform-store.js [hunk 1]
```diff
@@ -468,18 +468,17 @@ export class PlatformStore {
    * @param {string} serviceId
    */
   reportPlatformFailure(serviceId) {
-    if (this.failedPlatforms_.indexOf(serviceId) == -1) {
+    if (serviceId === this.getLocalPlatform().getServiceId()
+      && this.fallbackEntitlement_) {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/platform-store.js]: What would happen if the `fallbackEntitlement` is successful, but other services have not finished yet? Will we execute the "local" renderer and send "local" pingback even though some other services might return with an actual positive response? Or does "renderer" and "pingback" wait for other services anyway?

Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/platform-store.js]: Well in theory its just filling local with a proxied Entitlement and doesn't change anything with `selectPlatform` flow. But will cross check and see if we have a test for that

**Comment by prateekbh** [comment]
@dvoytenko @jpettitt it indeed unblocks in favor of local always, not merging this right now.
Working on a better fix

**Pull request merged**